flink-yarn/src/main/java/org/apache/flink/yarn/configuration/YarnDeploymentTarget.java

```diff
@@ -30,9 +30,11 @@
 /** A class containing all the supported deployment target names for Yarn. */
 @Internal
 public enum YarnDeploymentTarget {
-    PER_JOB("yarn-per-job"),
     SESSION("yarn-session"),
-    APPLICATION("yarn-application");
+    APPLICATION("yarn-application"),
+
+    @Deprecated
+    PER_JOB("yarn-per-job");
```

Comment:
why are you changing the order?

Reply:
Just as in the documentation it felt right to have the deprecated one at the bottom.